### PR TITLE
Phase 1 updates

### PR DESCRIPTION
**Pull request by sduffy77**

### CODE REVIEW

File: src/Extensions/StreamExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            formDataStream.Write(lineBreakBytes, 0, lineBreakBytes.Length);
+
+            if (File.Exists(formModel.FilePath))
+            { 
+                byte[] fileData = File.ReadAllBytes(formModel.FilePath);
+
+                // Write the file data directly to the Stream, rather than serializing it to a string.
+                formDataStream.Write(fileData, 0, fileData.Length);
+                formDataStream.Write(lineBreakBytes, 0, lineBreakBytes.Length);
+            }
```
Comment (deevezina) [inline src/Extensions/StreamExtensions.cs:52]: What if file doesn't exist?

File: src/ProjectDirectorClient.cs [hunk 2]
```diff
@@ -1,53 +1,414 @@
...
+            return returnValue;
+        }
+
+        /// <summary>
+        /// Finds the projects by name.
+        /// </summary>
+        /// <param name="projectNameFilter">The project name filter.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// Returns a <see cref="List{ProjectDirectorProjectModel}" /> objects.
```
Comment (deevezina) [inline src/ProjectDirectorClient.cs:92]: List{ProjectMicroModel}

File: src/ProjectDirectorClient.cs [hunk 3]
```diff
@@ -1,53 +1,414 @@
...
+        #region Projects
+
         /// <summary>
         /// Retrieves the projects from Project Director.
         /// </summary>
+        /// <param name="pageSize">Size of the page.</param>
+        /// <param name="orderBy">The order by.</param>
+        /// <param name="cancellationToken">Contains the cancellation token.</param>
+        /// <returns>
+        /// Returns a <see cref="List{ProjectDirectorProjectModel}" /> objects.
```
Comment (deevezina) [inline src/ProjectDirectorClient.cs:44]: List{ProjectMicroModel}

File: src/Extensions/StreamExtensions.cs (hunk 1 above)
Comment (sduffy77) [reply src/Extensions/StreamExtensions.cs:52]: There are checks in the endpoint methods for File.Exists before this extension method is called. 
Perhaps we should also check up front in the extension method and throw a new exception if it doesn't exist here. I'll update.

File: src/ProjectDirectorClient.cs (hunk 2 above)
Comment (sduffy77) [reply src/ProjectDirectorClient.cs:92]: Good catch, this was changed after I added the method documentation.

File: src/ProjectDirectorClient.cs (hunk 3 above)
Comment (sduffy77) [reply src/ProjectDirectorClient.cs:44]: Good catch, this was changed after I added the method documentation.

**Pull request merged**